### PR TITLE
Container type support

### PR DESCRIPTION
**Pull request by soualid**
Implements container type support as defined here:  http://mesos.apache.org/documentation/latest/container-image/

The container type is detected automatically within the JSON template, or can be defined through the plugin configuration, it's defaulted to the legacy `DOCKER` value if not defined (which was the value defined by previous versions of the plugin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Marathon Plugin for Jenkins [![Build Status](https://jenkins.mesosphere.com/service/jenkins/buildStatus/icon?job=marathon-plugin-publish-master)](https://jenkins.mesosphere.com/service/jenkins/job/marathon-plugin-publish-master/)
+# Marathon Plugin for Jenkins
```
Comment (colin-msphere) [inline README.md]: The new build icon can be found here: https://jenkins.mesosphere.com/service/jenkins/buildStatus/icon?job=marathon-jenkins-plugin/marathon-plugin-publish-master

**Comment by colin-msphere** [comment]
@mesosphere-ci ok to test

**Comment by colin-msphere** [comment]
Errors from CI:

```
Tests in error: 
com.mesosphere.velocity.marathon.impl.MarathonBuilderImplTest.testSimpleContainerTypeDetection(com.mesosphere.velocity.marathon.impl.MarathonBuilderImplTest)
  Run 1: MarathonBuilderImplTest.testSimpleContainerTypeDetection:143 » NullPointer
```

File: README.md (hunk 1 above)
Comment (soualid) [reply README.md]: Thanks, just updated it.

**Comment by colin-msphere** [comment]
Thanks for the update.

**Pull request merged**